business/authentication/openid_auth_controller.go

```diff
@@ -367,12 +367,14 @@ func (c OpenIdAuthController) redirectToAuthServerHandler(w http.ResponseWriter,
 		return
 	}
 
+	guessedKialiURL := httputil.GuessKialiURL(r)
+	secureFlag := conf.IsServerHTTPS() || strings.HasPrefix(guessedKialiURL, "https:")
```

Comment:
Since this is repeated a bunch of places now can it just be its own function? Something like `IsSecureCookie(...)`?